algoliasearch-core/src/main/java/com/algolia/search/models/rules/ConsequenceQuery.java

```diff
@@ -31,29 +37,118 @@ public ConsequenceQuery setEdits(List<Edit> edits) {
     this.edits = edits;
     return this;
   }
+
+  String getQueryString() {
+    return queryString;
+  }
+
+  ConsequenceQuery setQueryString(String queryString) {
+    this.queryString = queryString;
+    return this;
+  }
+
+  private List<Edit> edits;
+  private String queryString;
+}
+
+/**
+ * Custom serializer to handle polymorphism/legacy of "query" attributes in ConsequenceQuery
+ *
+ * <p>Example:
+ *
+ * <pre>{@code
+ *  // query string
+ *  "query": "some query string"
+ *
+ *  // remove attribute (deprecated)
+ *  "query": {"remove": ["query1", "query2"]}
+ *
+ *  // edits attribute
+ *  "query": {
+ *     "edits": [
+ *        { "type": "remove", "delete": "old" },
+ *        { "type": "replace", "delete": "new", "insert": "newer" }
+ *     ]
+ * }
+ * }</pre>
+ */
+class ConsequenceQuerySerializer extends JsonSerializer<ConsequenceQuery> {
+  @Override
+  public void serialize(ConsequenceQuery value, JsonGenerator gen, SerializerProvider serializers)
+      throws IOException {
+    /*
+     * Consequence query edits will override regular "query" - both can't be set at the same time
+     * https://www.algolia.com/doc/api-reference/api-methods/save-rule/#method-param-query
+     * */
+    if (!AlgoliaUtils.isNullOrEmptyWhiteSpace(value.getQueryString())) {
+      gen.writeString(value.getQueryString());
```

Comment:
You may want to migrate `value.getQueryString()` in its own temporary: it's probably negligible in term of performances but for readability, it would help at least.

Reply:
> You may want to migrate `value.getQueryString()` in its own temporary: it's probably negligible in term of performances but for readability, it would help at least.

Could you elaborate? i don't get it.

Reply:
I was suggesting something like this:

```java
String query = value.getQueryString();

 if (!AlgoliaUtils.isNullOrEmptyWhiteSpace(query)) {
      gen.writeString(query);
} else {
     // ...
}
```